### PR TITLE
[17825] prevent composer from focusing on opening chat with edit/reply

### LINKED ISSUE #17825: Composer focuses when re-entering with unfinished edit/reply
# Bug Report

When editing/replying to a message and leaving the chat without finishing and re-entering the chat, the composer is focused and the keyboard appears. It should behave as it does when leaving the chat with an unfinished message (not editing) - not focused. 

NOTE for the devs, most likely the issue is in the `...chat.composer.effects/use-edit` and `.../use-reply` effects and we need a condition for re-entering the screen.

https://github.com/status-im/status-mobile/assets/21865759/6c0e7137-1686-48eb-8783-6b47ba0382a0

#### Expected behavior

The input is not focused when re-entering the screen.

#### Actual behavior

The input is focused when re-entering the screen.

### Reproduction

1. Start editing/replying to a message
2. Hide the composer
3. Go back to the chat list
4. Re-enter the chat
5. The composer is focused

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by ibrkhalil**
fixes #17825 

### Summary

Prevents composer from auto-focusing when opening a chat with edit/reply

status: ready

### CODE REVIEW

**Comment by clauxx** [comment]
Thanks for the PR! So from what I see we're not focusing when editing at all. When the user starts editing a message, the input should be focused. We just shouldn't focus when re-entering the screen with an on-going edit.

**Comment by ibrkhalil** [comment]
> Thanks for the PR! So from what I see we're not focusing when editing at all. When the user starts editing a message, the input should be focused. We just shouldn't focus when re-entering the screen with an on-going edit.

Thanks for the quick review, Looking into it.

**Comment by ibrkhalil** [comment]
> Thanks for the PR! So from what I see we're not focusing when editing at all. When the user starts editing a message, the input should be focused. We just shouldn't focus when re-entering the screen with an on-going edit.

Should be done now .. Thanks!

File: src/status_im2/contexts/chat/composer/effects.cljs [hunk 1]
```diff
@@ -106,12 +106,13 @@
...
    {:keys [text-value saved-cursor-position]}
-   {:keys [edit]}]
+   {:keys [edit]}
+   message-list-rendered?]
   (rn/use-effect
    (fn []
      (let [edit-text        (get-in edit [:content :text])
            text-value-count (count @text-value)]
-       (when (and edit @input-ref)
+       (when (and edit @input-ref @message-list-rendered?)
```
Comment (clauxx) [inline src/status_im2/contexts/chat/composer/effects.cljs]: I think this needs a more self-explanatory name? smth like `composer-layout-done?`?

**Review by clauxx — APPROVED** [review]
Looks good! 👍

File: src/status_im2/contexts/chat/composer/effects.cljs (hunk 1 above)
Comment (ibrkhalil) [reply src/status_im2/contexts/chat/composer/effects.cljs]: Done

**Comment by churik** [comment]
### ISSUE 1: empty content of editng message is erased from composer when reopening chat

**Steps:**
1) send a message to 1-1 chat
2) tap back 2 times in order to get to message home view
3) open the same chat view again and check the composer

**Expected result:**

text is stayed in composer

**Actual result:**

https://github.com/status-im/status-mobile/assets/4557972/260e0e7f-ee86-4d28-9701-ea516b18af25



**OS:** IOS, Android

**Comment by ibrkhalil** [comment]
> ### ISSUE 1: еру content of editшng message is erased from composer when reopening chat
> **Steps:**
> 
> 1. send a message to 1-1 chat
> 2. tap back 2 times in order to get to message home view
> 3. open the same chat view again and check the composer
> 
> **Expected result:**
> 
> text is stayed in composer
> 
> **Actual result:**
> 
>  FILE.2023-11-14.17.43.53.mp4 
> **OS:** IOS, Android

Thank you so much for the review!
Should be fixed now.

**Comment by churik** [comment]
blocking until https://github.com/status-im/status-mobile/pull/17785 merge - should be rebased and rechecked after, as composer behavior drastically changed there

**Comment by churik** [comment]
@ibrkhalil 
is it not a part of https://github.com/status-im/status-mobile/pull/17785 ?
If it is, we can close it, thank you!

**Pull request closed without merging**